src/components/MerchantPayout/MerchantPayoutScreen.tsx

```diff
@@ -78,42 +82,64 @@ export const MerchantPayoutScreen: FunctionComponent<NavigationFocusInjectedProp
   const [isScanningEnabled, setIsScanningEnabled] = useState(true);
   const [vouchers, setVouchers] = useState<Voucher[]>([]);
   const [merchantCode, setMerchantCode] = useState("");
+  const [voucherStatus, setVoucherStatus] = useState<VoucherStatus>({
+    status: "VALID"
+  });
 
   useEffect(() => {
     if (isFocused) {
       setIsScanningEnabled(true);
     }
   }, [isFocused]);
 
+  const onCancel = (): void => {
+    setVoucherStatus({
+      status: "VALID"
+    });
+    setIsScanningEnabled(true);
+  };
+
   const onCheckVoucher = async (input: string): Promise<void> => {
     try {
       setIsScanningEnabled(false);
+      setVoucherStatus({ status: "CHECKING" });
       const serialArr = vouchers.map(voucher => voucher.serial);
       validateVoucherCode(input, serialArr);
       setVouchers([...vouchers, { serial: input, denomination: 2 }]);
       Vibration.vibrate(50);
-      setIsScanningEnabled(true);
+      //TODO: Send to API -> Valid, for now timeout
+      setTimeout(() => {
+        console.log("Sending to API");
+        setVoucherStatus({ status: "VALID" });
+        setIsScanningEnabled(true);
+      }, 1000);
     } catch (e) {
       setIsScanningEnabled(false);
-      Alert.alert(
-        "Error",
-        e.message || e,
-        [
-          {
-            text: "Dismiss",
-            onPress: () => setIsScanningEnabled(true)
-          }
-        ],
-        {
-          onDismiss: () => setIsScanningEnabled(true) // for android outside alert clicks
-        }
-      );
+      setVoucherStatus({ status: "INVALID", errorMessage: e.message });
     }
   };
 
   const onBarCodeScanned: BarCodeScannedCallback = event => {
     if (isFocused && isScanningEnabled && event.data) {
       onCheckVoucher(event.data);
+    } else {
+      setVoucherStatus({
+        status: "INVALID",
+        errorMessage: "Please try Scanning again or enter voucher manually",
+        errorTitle: "Error scanning"
+      });
+    }
+  };
+
+  const onVoucherCodeSubmit = (voucherCode: string): void => {
+    if (isFocused && voucherCode) {
+      onCheckVoucher(voucherCode);
+    } else {
+      setVoucherStatus({
+        status: "INVALID",
+        errorMessage: "Please log an appeal request",
+        errorTitle: "Error during input"
+      });
```

Comment:
In this case the error should be that no voucher code was entered. Also what's the reason for the `isFocused` check in this case?

Reply:
It was on the camera scanning screen for NRIC so we thought it was important. But we can remove!